tsdb/engine/tsm1/compact_test.go

```diff
@@ -1535,7 +1535,7 @@ func TestDefaultPlanner_Plan_Min(t *testing.T) {
 		}, tsdb.DefaultCompactFullWriteColdDuration,
 	)
 
-	tsm := cp.Plan(time.Now())
+	tsm, _ := cp.Plan(time.Now())
```

Comment:
Is there an existing test for the case when acquiring the lock fails where we could assert that the returned len is what we expect?

Reply:
There is not such a case, and I can't see a reasonable way to write one that won't be another flaky test.  The condition only occurs when the planner is running in parallel and encounters a race condition.  I have added checks on the test suite to ensure that the result is in the straightforward synchronous case.

Reply:
We could look at the code Monday to see if you have a good idea that eluded me...

Reply:
No need, just wanted to ask in case it was low-hanging fruit